include/deal.II/lac/precondition.h

```diff
@@ -476,6 +476,24 @@ class PreconditionJacobi : public PreconditionRelaxation<MatrixType>
 {
 public:
   /**
+   * AdditionalData to store relaxation parameter.
+   */
+  struct AdditionalData : public PreconditionRelaxation<MatrixType>::AdditionalData
```

Comment:
do you mean just using
```
typedef typename PreconditionRelaxation<MatrixType>::AdditionalData AdditionalData
```
?

I think that's all what is needed for `MGSmootherPrecondition` to work (just checked it).